subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -0,0 +1,716 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <bluetooth/mesh/models.h>
+#include <sys/byteorder.h>
+#include "model_utils.h"
+#include "mesh/net.h"
+#include "mesh/access.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#define LOG_MODULE_NAME bt_mesh_scene_srv
+#include "common/log.h"
+
+#define SCENE_PAGE_SIZE SETTINGS_MAX_VAL_LEN
+
+struct __packed scene_data {
+	uint8_t len;
+	uint8_t elem_idx;
+	uint16_t id;
+	uint8_t data[];
+};
+
+static char *scene_path(char buf[9], uint16_t scene, bool vnd, uint8_t page)
+{
+	sprintf(buf, "%x/%c%x", scene, vnd ? 'v' : 's', page);
+	return buf;
+}
+
+static void scene_status_encode(struct bt_mesh_scene_srv *srv,
+				struct net_buf_simple *buf,
+				enum bt_mesh_scene_status status)
+{
+	int64_t now = k_uptime_get();
+
+	bt_mesh_model_msg_init(buf, BT_MESH_SCENE_OP_STATUS);
+	net_buf_simple_add_u8(buf, status);
+
+	if (now < srv->transition_end) {
+		net_buf_simple_add_le16(buf, srv->prev);
+		net_buf_simple_add_le16(buf, srv->curr);
+		net_buf_simple_add_u8(buf, model_transition_encode(
+						   srv->transition_end - now));
+	} else {
+		net_buf_simple_add_le16(buf, srv->curr);
+	}
+}
+
+static int scene_status_send(struct bt_mesh_scene_srv *srv,
+			     struct bt_mesh_msg_ctx *ctx,
+			     enum bt_mesh_scene_status status)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCENE_OP_STATUS,
+				 BT_MESH_SCENE_MSG_MAXLEN_STATUS);
+
+	scene_status_encode(srv, &buf, status);
+
+	return model_send(srv->mod, ctx, &buf);
+}
+
+static void handle_get(struct bt_mesh_model *mod, struct bt_mesh_msg_ctx *ctx,
+		       struct net_buf_simple *buf)
+{
+	struct bt_mesh_scene_srv *srv = mod->user_data;
+
+	(void)scene_status_send(srv, ctx, BT_MESH_SCENE_SUCCESS);
+}
+
+static void scene_set(struct bt_mesh_scene_srv *srv, uint16_t scene)
+{
+	if (srv->curr == scene) {
+		return;
+	}
+
+	srv->curr = scene;
+	if (scene) {
+		bt_mesh_model_data_store(srv->mod, false, "s", &srv->curr,
```

Comment:
Can `"s"` be in a macro definition to make it clear what it means and what it is used for?